File: ios/Detonator/Sources/Detonator/Module/Module.swift
import Foundation

open class Module: NSObject {
    public let detonator: Detonator
    
    required public init(_ detonator: Detonator) {
        self.detonator = detonator
    }
    
    open func setUp() -> Void {
        preconditionFailure("This method must be overriden.")
    }
}

File: android/Detonator/src/main/java/com/iconshot/detonator/module/Module.java
package com.iconshot.detonator.module;

import com.iconshot.detonator.Detonator;

public abstract class Module {
    protected Detonator detonator;

    public Module(Detonator detonator) {
        this.detonator = detonator;
    }

    public abstract void setUp();
}
